CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+	this.object = xmlQueryUrl + url.match( objectID )[ 1 ] + "/show/xml";
+  } else this.object = "";
+  if ( itemUrlHttp ) this.item = xmlQueryUrl + itemUrlHttp.match( objectID )[ 1 ] + "/show/xml";
+  else if ( this.object && url.match( itemID ) ) this.item = xmlQueryUrl + url.match( itemID )[ 1 ] + "/show/xml";
+  else this.item = "";
+}
+
+function scrape( doc, url ) {
+  var item = new Zotero.Item( detectWeb( doc, url ) );
+  url = doc.documentURI;
+  var xmlUrl = new fetchXmlUrl( doc, url );
+
+  /* Mappings for XML and Web (CDM>5)
+   * The scraper only scrapes fields specified in the mapping table.
+   * Enter mappings as [zoteroField, CDMfield];
+   * note CDM allows customers to configure fields
+   * so these are inconsistently implemented.
+   */
+  var mapping = [
+			[ 'title', 'title' ],
+			[ 'creator', 'creato' ],
+			[ 'creator2', 'creata' ],
+			[ 'contributor', 'contri' ],
+			[ 'abstractNote', 'descri' ],
+			[ 'language', 'langua' ],
+			[ 'language2', 'languag' ],
+			[ 'archiveLocation', 'identi' ], // On CDM this is 'Identifier'
+			[ 'archiveLocation2', 'locala' ],
+			[ 'archive', 'source' ],
+			[ 'collection', 'collec' ],
+			[ 'collection2', 'relati' ],
+			[ 'publisher', 'publis' ], // Omitted by manuscript item type
+			[ 'rights', 'rights' ],
+			[ 'subject', 'subject' ], // Scraped into tags
+			[ 'subject2', 'subjea' ],
+			[ 'subject3', 'subjeb' ],
+			[ 'subject4', 'subjec' ],
+			[ 'person', 'person' ],
+			[ 'medium', 'type' ],
+			[ 'format', 'format' ],
+			[ 'date', 'date' ],
+			[ 'date2', 'dated' ],
+			[ 'place', 'covera' ],
+			[ 'place2', 'coverab' ],
+			[ 'period', 'period' ],
+		  ];
+
+  /* Mappings for Web (CDM<6)
+   * CDM < 6 does not provide standard field names.
+   * The scraper scrapes the whole display table,
+   * and converts displayed fieldnames to camelCase
+   * to maximise the chance of a Zotero match.
+   * Hardcode additional matches here as [zoteroField, displayedField];
+   */
+
+  var legacyMapping = [ // Enter mappings as [zoteroField,camelizedCDMField]
+			[ "place", "town" ],
+			[ "abstractNote", "description" ],
+			[ "subject2", "keywords" ],
+			[ "subject3", "subjectKeywords" ],
+			[ "date2", "workDate" ],
+			[ "creator", "bookCreator" ],
+			[ "rights", "rights--UsageStatement" ],
+			[ "contributor2", "photographer" ],
+			[ "title2", "bookTitle" ],
+		  ];
+
+  var docTitle = ( function() {
+	if ( doc.title == "ContentDM" ) return url.substring( url.indexOf( '://' ) + 3, url.indexOf( '/cdm/' ) ).replace( /^\s+|\s+$/g, '' );
+	else return doc.title.replace( /^\s+|\s+$/g, '' );
+  }() );
+
+  item.libraryCatalog = ( function() {
+	var catalog = fetchXPathText( doc, '//*[@id="breadcrumb_top_content"]/a[2]' );
+	if ( catalog ) return catalog;
+	else if ( docTitle.match( '::' ) ) return docTitle.substring( doc.title.indexOf( '::' ) + 3 ); // Fallback: use second part of webpage title
+	else return docTitle;
+  }() );
```

Comment:
I'll let @adam3smith decide this, but in general, we just let this be the URL host. Given that this is an online catalog, it's both informative for the user and it helps us with debugging.